directord/drivers/grpcd.py

```diff
@@ -405,10 +422,50 @@ def wait_for_connection(connectivity):
         compression_type = grpc.Compression.Gzip
         if not self.compression:
             compression_type = grpc.Compression.NoCompression
-        self.channel = grpc.insecure_channel(
-            f"{self.server_address}:{self.server_port}",
-            compression=compression_type,
-        )
+
+        if self.secure:
+            ssl_cert = None
+            ssl_key = None
+            if not os.path.exists(self.ssl_ca):
+                raise Exception(
+                    "SSL is enabled but CA file does not exist "
+                    f"({self.ssl_ca})"
+                )
+            with open(self.ssl_ca, "rb") as ca_file:
+                ca_cert = ca_file.read()
+
+            if (
+                not self.ssl_cert
+                or not os.path.exists(self.ssl_cert)
+                or not self.ssl_key
+                or not os.path.exists(self.ssl_key)
+            ):
+                self.log.warning(
+                    "Client SSL Cert or Key do not exist. "
+                    "Skipping configuration"
```

Comment:
this should either raise an error, or continue from the line #456

Reply:
It's not an error condition necessarily. If requiring client auth is set to false, we just need the CA for TLS support w/o auth.